src/NServiceBus.Core/Hosting/Helpers/AssemblyScanner.cs

```diff
@@ -111,10 +111,12 @@ static List<Type> GetHandlerMessageTypes(List<Type> list)
                     {
                         var messageType = @interface.GetGenericArguments()[0];
                         foundMessageTypes.Add(messageType);
+
+                        foundMessageTypes.AddRange(messageType.Assembly.GetTypes().Where(t=> messageType.IsAssignableFrom(t)));
```

Comment:
It was just a hack :) 

Regarding base type, in this case it the base type we find so we need to go in the other direction? (types inheriting from the type found). What's the best way to do that?